app/models/history.rb

```diff
@@ -463,30 +466,41 @@ def self.clear_follow_up_flag(history)
     to_text = field[:new_value].blank? ? 'blank' : "\"#{field[:new_value]}\""
 
-    comment = "#{creator} #{verb} #{field[:name]} from #{from_text} to #{to_text}"
-    comment += compose_explanation(history, field)
+    comment = "User #{verb} #{field[:name]} from #{from_text} to #{to_text}"
+    comment += compose_explanation(history)
     comment += history[:note] unless history[:note].blank?
     comment += '.'
     comment += " Reason: #{history[:reason]}" unless history[:reason].blank?
     comment
   end
 
-  private_class_method def self.compose_explanation(history, field)
-    if history[:household_status] == :patient && history[:propagation] == :group
-      " and chose to update this #{field[:type]} for all household members"
-    elsif history[:household_status] == :patient && history[:propagation] == :group_cm
-      " and chose to update this #{field[:type]} for household members under continuous exposure"
-    elsif history[:household_status] != :patient && history[:propagation] == :group
-      " because User updated #{field[:name]} for another member in this monitoree's household and chose to update this
-        #{field[:type].nil? ? 'field' : field[:type]} for all household members"
-    elsif history[:household_status] != :patient && history[:propagation] == :group_cm
-      " because User updated #{field[:name]} for another member in this monitoree's household and chose to update this
-        #{field[:type].nil? ? 'field' : field[:type]} for household members under continuous exposure"
+  private_class_method def self.compose_explanation(history)
+    if history[:initiator_id] == history[:patient].id && history[:propagation] == :group
+      ' and applied that change to all household members'
+    elsif history[:initiator_id] == history[:patient].id && history[:propagation] == :group_cm
+      ' and applied that change to household members under continuous exposure'
+    elsif history[:initiator_id] == history[:patient].responder_id && history[:propagation] == :group
+      " by making that change for monitoree's head of household (Sara Alert ID: #{history[:initiator_id]})"\
+      ' and for all household members'
+    elsif history[:initiator_id] == history[:patient].responder_id && history[:propagation] == :group_cm
+      " by making that change for monitoree's head of household (Sara Alert ID: #{history[:initiator_id]})"\
+      ' and for household members under continuous exposure'
+    elsif history[:initiator_id] != history[:patient].id && history[:initiator_id] == history[:patient].responder_id
+      " by making that change for monitoree's head of household (Sara Alert ID: #{history[:initiator_id]})"\
+      ' and for this monitoree'
+    elsif history[:initiator_id] != history[:patient].id && history[:propagation] == :group
+      " by making that change for a household member (Sara Alert ID: #{history[:initiator_id]})"\
+      ' and applied that change to all household members'
```

Comment:
I can't figure out how to trigger this case. Far as I can tell, the only way to get the `group` option is via a `bulk_update`. But if you do a bulk_update, the person making that change and applying it to their household is always going to be the HOH, so the `initiator_id` is going to be equal to the `patient.id`, since if we update on a member of a household who _isn't_ the HOH, we can't apply that change to the rest of the household via `bulk_update`. At least that's what I'm seeing, but do you disagree? Is there a way you have for recreating this case?

Reply:
Yeah, It looks like you're right. The logic is carefully designed in bulk_edit to make sure this case never gets hit. Removing.